BinanceExchange.API/Trade/BinTrade.cs

```diff
@@ -0,0 +1,191 @@
+﻿using System;
+using System.Threading.Tasks;
+using BinanceExchange.API.Client;
+using BinanceExchange.API.Enums;
+using BinanceExchange.API.Models.Request;
+
+//"dynamic" requires a reference to Microsoft.CSharp
+
+namespace BinanceExchange.API.Client.Trade
+{
+    public class MarginTrade
+    {
+        /// <summary>
+        /// Send isolated account buy command
+        /// </summary>
+        /// <param name="client"></param>
+        /// <param name="symbol">the trading symbol (BTCUSDT...)</param>
+        /// <param name="asset">for which the buy power is queried</param>
+        /// <returns></returns>
+        public static async Task<decimal> MaxBuyPowerInAsset(BinanceClient client, string symbol, string asset)
+        {
+            dynamic accountInfo = await client.QueryIsolatedMarginAccountInfo(new IsolatedMarginAccountInfoRequest()
+            {
+                Symbols = symbol,
+                TimeStamp = DateTime.Now.Ticks
+            });
+
+            dynamic x = accountInfo.assets[0];
+
+            // query the account free asset coins:
+            decimal freeCoins = 0m;
+            if(asset == ((dynamic)x).baseAsset.asset.Value)
+            {// the requested asset is the base
+                freeCoins = decimal.Parse(((dynamic)x).baseAsset.free.Value);
+            }
+            else if(asset == ((dynamic)x).quoteAsset.asset.Value)
+            {// the requested asset is the one used for quotes
+                freeCoins = decimal.Parse(((dynamic)x).quoteAsset.free.Value);
+            }
+            else
+            {
+                throw new Exception("the rquested asset info was not founf under the specified accout");
+            }
+
+            var maxBorrowAsset = await client.QueryMaxBorrow(new MaxBorrowRequest()
+            {
+                Asset = asset,
+                IsolatedSymbol = symbol,
+            });
+
+            return maxBorrowAsset.amount + freeCoins;
+        }
+
+        private static async Task<bool> _BuySellCommand(BinanceClient client, string symbol, OrderSide buySell, SideEffectType sideEffectType, string isIsolated, decimal price, decimal quant)
+        {
+            var createIsolatedOrder = await client.CreateIsolatedOrder(new CreateIsolatedOrderRequest()
+            {
+                Price = price,
+                Quantity = quant,
+                Side = buySell,
+                Symbol = symbol,
+                Type = OrderType.Limit,
+                NewOrderResponseType = NewOrderResponseType.Full,
+                TimeInForce = TimeInForce.GTC,
+                IsIsolated = isIsolated,
+                SideEffectType = sideEffectType,
+                TimeStamp = DateTime.UtcNow.Ticks
+            });
+
+            return (true);
```

Comment:
Can we check the result type instead to determine if this was success? Returning true without even looking at the order doesn't seem right. Perhaps return an ID or something instead?

Reply:
good point - OrderId it is